rpcs3/Emu/RSX/RSXThread.cpp

```diff
@@ -2491,6 +2491,12 @@ namespace rsx
 		m_invalidated_memory_range.invalidate();
 	}
 
+	void thread::invalidate_memory_range(const utils::address_range& range, rsx::invalidation_cause cause)
+	{
+		std::lock_guard lock(m_mtx_task);
+		on_invalidate_memory_range(range, cause);
```

Comment:
This will crash drivers. It is unsafe to execute write invalidation on a thread that is not rsx core. It will definitely crash on OGL and may cause trouble on VK.